common/types/int_test.go

```diff
@@ -75,12 +93,22 @@ func TestInt_ConvertToNative_Int64(t *testing.T) {
 }
 
 func TestInt_ConvertToNative_Json(t *testing.T) {
-	val, err := Int(4147483648).ConvertToNative(jsonValueType)
+	// Value less than int32.
+	val, err := Int(maxIntJSON).ConvertToNative(jsonValueType)
+	if err != nil {
+		t.Error(err)
+	} else if !proto.Equal(val.(proto.Message),
+		&structpb.Value{Kind: &structpb.Value_NumberValue{NumberValue: 9007199254740991.0}}) {
+		t.Errorf("Got '%v', expected a json number for a 32-bit int", val)
+	}
+
+	// Value greater than max int32.
```

Comment:
"Value converts to a JSON decimal string".

(Hmm, +/- 2^53 actually *can* be represented accurately as a double, in case you want to expand the range in each direction by one.  But integer maxes are usually one less than a power of 2.)

Reply:
The `maxIntJSON` is copied from the EcmaScript 6 standard, so 2^53-1